src/main/java/com/aws/greengrass/clientdevices/auth/configuration/InfrastructureConfiguration.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.configuration;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.util.Coerce;
+import lombok.Value;
+
+/**
+ * Represents client device infrastructure configuration.
+ * </p>
+ * NOTE: currently we're shoving some unrelated things under the `performance` key.
+ * Things like maxActiveAuthTokens and refresh periods should be grouped separately.
+ * <p>
+ * |---- configuration
+ * |    |---- performance:
+ * |          |---- cloudRequestQueueSize: "..."
+ * |          |---- maxConcurrentCloudRequests: [...]
+ * </p>
+ */
+@Value
+public final class InfrastructureConfiguration {
+    public static final int DEFAULT_WORK_QUEUE_DEPTH = 100;
+    public static final int DEFAULT_THREAD_POOL_SIZE = 1;
+
```

Comment:
I don't love this configuration - might make sense to re-organize a bit. But for now, we'll keep it how it is.

Note: I didn't want to change the existing SessionConfig, so this performance key is split between the two config classes. I don't love it, but I also wanted to keep this PR as contained as possible.